widgetastic_manageiq/__init__.py

```diff
@@ -825,6 +825,115 @@ def workaround_save_issue(self):
             "{}._handlers.change.map(function(handler) {{ handler() }});".format(self.item_name))
 
 
+class SettingsGroupSubmenu(NavDropdown):
+    """
+    Widget for the Group 'Dropdown' menu that allows a user assigned to multiple groups
+    to change their active group dynamically Multiple group functionality added in 5.9
+    """
+    GROUP_SUBMENU = './/ul[contains(@class, "dropdown-menu scrollable-menu")]'
+
+    def __init__(self, parent, locator, logger=None):
+        Widget.__init__(self, parent, logger=logger)
+        self.locator = locator
+
+    def __locator__(self):
+        return self.locator
+
+    @property
+    def is_displayed(self):
+        return self.parent.expanded
+
+    @property
+    def expandable(self):
+        try:
+            self.browser.element('./.[contains(@class, "dropdown-submenu")]', parent=self)
+        except NoSuchElementException:
+            return False
+        else:
+            return True
+
+    @property
+    def expanded(self):
+        if not self.expandable:
+            return False
+        return self.browser.element(self.GROUP_SUBMENU, parent=self).is_displayed()
+
+    def expand(self):
+        """ Expand the Settings->Change Group submenu by hovering. """
+        if not self.expandable:
+            raise ValueError('{} is not expandable'.format(self.locator))
+
+        # Make sure the Settings menu is expanded
+        if not self.parent.expanded:
+            self.parent.expand()
+
+        if not self.expanded:
+            # Hover over 'Change Group'
+            self.move_to()
+            if not self.expanded:
+                raise Exception('Could not expand {}'.format(self.locator))
+            else:
+                self.logger.info('expanded')
+
+    def collapse(self):
+        if not (self.parent.is_displayed or self.expandable):
+            return
+
+        if self.expanded:
+            self.browser.move_to_element(self.parent)
+            if self.expanded:
+                raise Exception('Could not collapse {}'.format(self.locator))
+            else:
+                self.logger.info('collapsed')
+
+    @property
+    def text(self):
+        try:
+            return self.browser.text('./a', parent=self)
+        except NoSuchElementException:
+            return None
+
+    @property
+    def items(self):
+        if not self.expandable:
+            return [self.text]
+        return [
+            self.browser.text(element)
+            for element
+            in self.browser.elements('{}/li'.format(self.GROUP_SUBMENU), parent=self)]
+
+    def has_item(self, item):
+        return item in self.items
+
+    def item_enabled(self, item):
+        if not self.has_item(item):
+            raise ValueError('There is no such item {}'.format(item))
```

Comment:
Include `self.items` in exception message so we can see from a failure whether it was something very small like incorrect capitalization.

Something like 'Available SettingsGroupSubmenu items are {}`.format(self.items) - you'll notice I've stuck this into the dropdown widgets for the same reason.